app/src/main/java/org/mozilla/fenix/settings/wallpaper/Extensions.kt

```diff
@@ -25,3 +28,37 @@ fun List<Wallpaper>.groupByDisplayableCollection(): Map<Wallpaper.Collection, Li
 }.toMap().takeIf {
     it.isNotEmpty()
 } ?: mapOf(Wallpaper.DefaultCollection to listOf(Wallpaper.Default))
+
+/**
+ * The ideal scenario is to have 3 seasonal wallpapers and 2 classic, minding the order.
+ * But in case there are less than 3 or no seasonal wallpapers at all, they are replaced by classic.
+ * Edge cases where there are less then 6 items overall would return less items in the right order.
+ */
+fun List<Wallpaper>.groupForWallpaperOnboarding(): List<Wallpaper> {
+    val result = mutableListOf(Wallpaper.Default)
+
+    val classicWallpapers = mutableListOf<Wallpaper>()
+    val seasonalWallpapers = mutableListOf<Wallpaper>()
+    for (wallpaper in this) {
+        if (wallpaper == Wallpaper.Default) continue
+
+        if (wallpaper.collection.name == "classic-firefox") {
+            classicWallpapers.add(wallpaper)
+        } else {
+            seasonalWallpapers.add(wallpaper)
+        }
+    }
+
+    if (seasonalWallpapers.size < SEASONAL_WALLPAPERS_COUNT) {
+        result.addAll(seasonalWallpapers)
+        result.addAll(classicWallpapers.take((maxSize - 1) - seasonalWallpapers.size))
```

Comment:
```suggestion
        result.addAll(classicWallpapers.take((maxSize - 1) - seasonalWallpapers.size))
```
My initial reaction was to wonder where `maxSize` was coming from. I think this code would be clearer if we didn't alias `THUMBNAILS_SELECTION_COUNT` as `maxSize`, but I imagine you did this because the name will take up a lot of space, but I would argue that we can just move `.take()` onto the next line.